### PR TITLE
[language] Test generation log to file

### PR DESCRIPTION
**Pull request by tzakian**
Add ability for test generation to log to a file (or terminal if no log file is specified).

Also record the error along with the module that failed.

### CODE REVIEW

File: language/tools/test-generation/src/main.rs [hunk 1]
```diff
@@ -8,21 +8,36 @@ use std::env;
...
+    if let Some(log_file_path) = log_file {
+        let file = std::fs::OpenOptions::new()
+            .create(true)
+            .write(true)
+            .truncate(true)
+            .open(log_file_path)
+            .unwrap();
+        let flog = slog_json::Json::default(file).fuse();
+        let drain = slog_envlogger::new(flog);
+        let drain = std::sync::Mutex::new(drain).fuse();
```
Comment (shazqadeer) [inline language/tools/test-generation/src/main.rs]: Many lines in common between then and else branch.  Some refactoring would help.

**Comment by davidiw** [comment]
Could we use https://github.com/libra/libra/tree/master/common/logger just as well and that way if we change loggers for the system we do it in one place?

**Comment by tzakian** [comment]
@davidiw I hadn't taken a look at that so I'll take a look and see. At least at first blush it sounds like a good idea. I'll look into porting it over and updating tomorrow.

**Comment by davidiw** [comment]
Closing this out as its been sitting here for a looong time. Feel free to re-open.

**Pull request closed without merging**